tribe-ext-settings-import-export.php

```diff
@@ -253,76 +233,94 @@ function tribe_sie_process_settings_import() {
 				return;
 			}
 
-			$import_file = $_FILES['import_file']['tmp_name'];
-			$import_filename = $_FILES['import_file']['name'];
-
-			if ( empty( $import_file ) ) {
-				wp_die( __( 'Please upload a file to import.', 'tribe-ext-settings-import-export' ) );
+			/**
+			 * Export actions.
+			 * Process a settings export that generates a .json file of the shop settings.
+			 */
+			if ( ! empty ( $_POST['export'] ) ) {
+				$settings = get_option( 'tribe_events_calendar_options' );
+
+				// TEC - widget_tribe-events-list-widget
+				// PRO - widget_tribe-events-adv-list-widget
+				// PRO - widget_tribe-events-countdown-widget
+				// PRO - widget_tribe-mini-calendar
+				// PRO - widget_tribe-events-venue-widget
+				// PRO - widget_tribe-events-venue-widget
+				// CE  - tribe_community_events_options
+				// CE  - Tribe__Events__Community__Schemaschema_version
+
+				ignore_user_abort( true );
+				nocache_headers();
+				header( 'Content-Type: application/json; charset=utf-8' );
+				header( 'Content-Disposition: attachment; filename=tribe-settings-export-' . date( 'm-d-Y' ) . '.json' );
+				header( "Expires: 0" );
+				echo json_encode( $settings );
 			}
 
-			if ( ! empty( $import_filename ) ) {
-				$tmp = explode( '.', $import_filename );
-				$extension = end( $tmp );
-			}
+			/**
+			 * Import actions.
+			 * Process a settings import from a json file.
+			 */
+			if ( ! empty ( $_POST['import'] ) ) {
 
-			if ( ! isset ( $extension ) || $extension != 'json' ) {
-				wp_die( __( 'Please upload a valid .json file.', 'tribe-ext-settings-import-export' ) );
-			}
+				$import_file = $_FILES['import_file']['tmp_name'];
+				$import_filename = $_FILES['import_file']['name'];
+
+				if ( empty( $import_file ) ) {
+					wp_die( __( 'Please upload a file to import.', 'tribe-ext-settings-import-export' ), __( 'Import error', 'tribe-ext-settings-import-export' ), [ 'back_link' => true ] );
+				}
 
-			// Retrieve the settings from the file and convert the json object to an array.
-			$settings = json_decode( file_get_contents( $import_file ), true );
+				if ( ! empty( $import_filename ) ) {
+					$tmp = explode( '.', $import_filename );
+					$extension = end( $tmp );
```

Comment:
This works, but we could alternatively do:

```suggestion
					$extension = pathinfo( $import_filename, PATHINFO_EXTENSION );
```
(Shorter, explicit, avoids the cost of array manipulation and saves us a variable assignment.)

Reply:
Oooooh, this is noice!!! Thanks!